zephyr/lib/alloc.c

```diff
@@ -257,8 +257,10 @@ static void virtual_heap_free(void *ptr)
 	ptr = (__sparse_force void *)sys_cache_cached_ptr_get(ptr);
 
 	ret = vmh_free(heap, ptr);
-	if (ret)
+	if (ret) {
 		tr_err(&zephyr_tr, "Unable to free %p! %d", ptr, ret);
+		k_panic();
```

Comment:
again not sure I understand why this is needed. Should we panic in all error cases? Or at least in all cases where we cannot directly propagate the error up the stack? Maybe we should use or add an IPC notification for some such cases.